src/systems/world_exporter/WorldExporter.cc

```diff
@@ -0,0 +1,331 @@
+ * limitations under the License.
+ *
+ */
+
+#include <ignition/plugin/Register.hh>
+
+#include <ignition/gazebo/components/Geometry.hh>
+#include <ignition/gazebo/components/Link.hh>
+#include <ignition/gazebo/components/Material.hh>
+#include <ignition/gazebo/components/Model.hh>
+#include <ignition/gazebo/components/Name.hh>
+#include <ignition/gazebo/components/ParentEntity.hh>
+#include <ignition/gazebo/components/Pose.hh>
+#include <ignition/gazebo/components/Transparency.hh>
+#include <ignition/gazebo/components/Visual.hh>
+#include <ignition/gazebo/components/World.hh>
+#include <ignition/gazebo/EntityComponentManager.hh>
+#include <ignition/gazebo/Util.hh>
+
+#include <sdf/Visual.hh>
+#include <sdf/Mesh.hh>
+#include <sdf/Model.hh>
+
+#include <ignition/common/Material.hh>
+#include <ignition/common/MeshManager.hh>
+#include <ignition/common/Mesh.hh>
+#include <ignition/common/SubMesh.hh>
+
+#include <ignition/math/Matrix4.hh>
+
+#include <ignition/common/ColladaExporter.hh>
+
+#include "WorldExporter.hh"
+
+using namespace ignition;
+using namespace gazebo;
+using namespace systems;
+
+
+class ignition::gazebo::systems::WorldExporterPrivate
+{
+  /// \brief Has the world already been exported?.
+  private: bool exported{false};
+
+  /// \brief Exports the world to a mesh.
+  /// \param[_ecm] _ecm Mutable reference to the EntityComponentManager.
+  public: void Export(const EntityComponentManager &_ecm)
+  {
+    if (this->exported) return;
+
+    common::Mesh worldMesh;
+    std::vector<math::Matrix4d> subMeshMatrix;
+
+    _ecm.Each<components::World, components::Name>(
+      [&](const Entity /*& _entity*/,
+        const components::World *,
+        const components::Name * _name)->bool
+    {
+      worldMesh.SetName(_name->Data());
+      return true;
+    });
+
+    _ecm.Each<components::Visual,
+            components::Name,
+            components::Geometry,
+            components::Transparency>(
+    [&](const ignition::gazebo::Entity &_entity,
+        const components::Visual *,
+        const components::Name *_name,
+        const components::Geometry *_geom,
+        const components::Transparency *_transparency)->bool
+    {
+      std::string name = _name->Data().empty() ? std::to_string(_entity) :
+          _name->Data();
+
+      igndbg << "====================== Starting: " << name << " - "
+        << _entity << "  ====================== " << std::endl;
+
+      math::Pose3d localPose = gazebo::worldPose(_entity, _ecm);
+
+      common::MaterialPtr mat = std::make_shared<common::Material>();
+      auto material = _ecm.Component<components::Material>(_entity);
+      if (material != nullptr)
+      {
+        mat->SetDiffuse(material->Data().Diffuse());
+        mat->SetAmbient(material->Data().Ambient());
+        mat->SetEmissive(material->Data().Emissive());
+        mat->SetSpecular(material->Data().Specular());
+      }
+      mat->SetTransparency(_transparency->Data());
```

Comment:
According to [this](https://www.khronos.org/files/collada_spec_1_4.pdf) Collada spec, a transparency of 0 means fully transparent, but in [SDFormat <transparency>](http://sdformat.org/spec?ver=1.7&elem=visual#visual_transparency), a value of 0 means fully opaque. So, I think we need to do `mat->SetTransparency(1 - _transparency->Data());` here.

As a quick test, I ran the example file and imported the generated dae file into Blender and all the shapes were invisible. Making the change I suggested above fixed it. I also tried running one of the cave worlds, but in that case the changes I suggested did not fix the issue, making me think there's probably another place to set transparency when meshes are involved.

Reply:
I'm a little confused. How do you see transparency into blender? I'm seeing this:
![blender1](https://user-images.githubusercontent.com/42421541/104212950-8a38f600-5414-11eb-8f82-ba3fa9116499.png)

Reply:
I actually used blender 2.9 and this is how it shows up
![image](https://user-images.githubusercontent.com/206116/104247364-456d8900-542d-11eb-8418-a69667b29a4e.png)

In 2.79, switching to the Material mode 
![image](https://user-images.githubusercontent.com/206116/104247323-31298c00-542d-11eb-8194-ab93358d61e7.png)

shows the colors are not solid. 
![image](https://user-images.githubusercontent.com/206116/104247732-f2e09c80-542d-11eb-89ef-2ddc3138f69f.png)

Reply:
This is resolved in https://github.com/ignitionrobotics/ign-common/pull/152

Reply:
![opaque](https://user-images.githubusercontent.com/1587438/105216944-17143b80-5b08-11eb-8280-fb9b8e1f53b0.png)

Reply:
The fix in ign-common has been released.